scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,378 @@
+deprecated module should look like, using `"This file is no longer relevant"` as the (optional)
+<comment>.
+
+If the message is not explicitly used, `#create_deprecated_module` defaults to
+`"Auto-generated deprecation"`.
+If you wish there to be no comment, use `#create_deprecated_module path/To/DeletedFile.lean ""`.
+
+If `rename_to "Mathlib/Path/To/Rename.lean"` is present, then instead of copying over the imports
+from a deleted file, it uses `import Mathlib.Path.To.Rename`.
+
+Finally, if everything looks correct, adding a final `write` actually generates the file:
+```lean
+#create_deprecated_module path/To/DeletedFile.lean "This file is no longer relevant" write
+```
+-/
+syntax "#create_deprecated_module " str (ppSpace str)? (&" rename_to " str)? (&" write")? ppLine :
+  command
+
+/-- `processPrettyOneLine log msg` takes as input two strings `log` and `msg`.
+It expects `log` to be a line in the output of `git log --pretty=oneline`:
+it should look like `<hash> <PRdescr>`.
+
+It returns the pair `(<hash>, <msg> in <PRdescr> <hash> <diff of file wrt previous commit>)`,
+formatted as a collapsible message. In practice, `msg` is either `last modified` or `deleted`.
+it returns the pair `(<hash>, <msg> in <PRdescr> <hash> <diff of file wrt previous commit>)`,
+formatted as a collapsible message.
+-/
+def processPrettyOneLine (log msg fname : String) : IO (String × MessageData) := do
+  let hash := log.takeWhile (!·.isWhitespace)
+  let PRdescr := (log.drop hash.length).trim
+  let gitDiffCLI := s!"git diff {hash}^...{hash} -- {fname}"
+  let diff ← runCmd gitDiffCLI <|> pure s!"{hash}: Error in computing '{gitDiffCLI}'"
+  let diffCollapsed := .trace {cls := .str .anonymous s!"{hash}"} m!"{gitDiffCLI}" #[m!"{diff}"]
+  return (hash, m!"{msg} in " ++
+    .trace {cls := .str .anonymous ("_" ++ hash.take 7)} m!"{PRdescr}" #[diffCollapsed])
+
+/--
+`mkRenamesDict pct` takes as optional input a natural number.
+
+It computes the `git` status of the files at the current `HEAD` commit,
+comparing them with `master`.
+
+It returns a `HashMap` with keys the old names and values the new names of all the files that
+git considers renames with likelihood at least the input `pct`.
+
+If no input is provided, the default percentage is `100`.
+-/
+def mkRenamesDict (percent : Nat := 100) : IO (Std.HashMap String String) := do
+  let mut dict := ∅
+  let gitDiff ← runCmd s!"git diff --name-status origin/master...HEAD"
+  let lines := gitDiff.trim.splitOn "\n"
+  for git in lines do
+    -- If `git` corresponds to a rename, it contains `3` segments, separated by a
+    -- tab character (`\t`): `R%%`, `oldName`, `newName`.
+    let [pct, oldName, newName] := git.split (· == '\t') | continue
+    if pct.take 1 != "R" then
+      IO.println
+        s!"mkRenamesDict: '{pct}' should have been of the form Rxxx, denoting a `R`ename \
+          and a similarity percentage.\nFull git line: '{git}'"
+      continue
+    let pctNat := (pct.drop 1).toNat?.getD 0
```

Comment:
If `percent` is 0, but the output line has the form `Rsdfdsf <something> <something>`, this will count the file as renamed. This is a small bug and probably never occurs in practice... but it could be nicer to be robust here.
```suggestion
    let some pctNat := (pct.drop 1).toNat? | continue
```